### PR TITLE
Messagebar Component

### PR DESCRIPTION
**Pull request by richardtagger**
Adds the message bar component - awaiting sign off from @adrianmg 

## Pull request checklist

* [x] Component `README.md` file is up-to-date.
* [x] Component visual reference files are up-to-date.
* [x] Component is unit tested.
* [x] Somebody from the UX team has reviewed your changes.

### CODE REVIEW

**Comment by richardtagger** [comment]
@unindented @annr

**Comment by unindented** [comment]
Why did all your reference files change? Only your feature should have changed right? Did you use docker?

**Comment by richardtagger** [comment]
Not sure why that happened. Fixed

**Comment by richardtagger** [comment]
Ok updated to Adrian's spec. Ready for review

File: src/components/MessageBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  if (!props.actions) {
+    return null;
+  }
+  return (<div className="y-message-bar--actions">{props.actions}</div>);
+};
+
+const MessageBar: React.StatelessComponent<MessageBarProps> = (props) => {
+  const actions = getActions(props);
+  return (
+    <div className={`y-message-bar y-message-bar__type-${props.type}`} aria-label={props.ariaLabel}>
```
Comment (unindented) [inline src/components/MessageBar/index.tsx]: All our components can take a `className`. You can do something like this: https://github.com/Microsoft/YamUI/blob/958d59a11d6035debf675379c434077c1d2c28f4/src/components/Callout/callout.tsx#L135

Comment (richardtagger) [reply src/components/MessageBar/index.tsx]: Alright - is that documented?

File: src/components/MessageBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  WARNING: 'warning',
+  ERROR: 'error',
+});
+export type MessageBarType = Enum<typeof MessageBarType>;
+
+export interface MessageBarProps extends NestableBaseComponentProps {
+  ariaLabel: string;
+  type?: MessageBarType;
+  actions?: JSX.Element;
+  children: JSX.Element | string;
```
Comment (unindented) [inline src/components/MessageBar/index.tsx]: I think these should not be `JSX.Element`, but `React.ReactNode[]`.

File: src/components/MessageBar/message-bar.css [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+.y-message-bar__type-info {
+  border-left-color: $color__punkRock;
+}
+
+.y-message-bar__type-warning {
+  border-left-color: $color__noon;
+}
+
+.y-message-bar__type-error {
+  border-left-color: $color__angel;
```
Comment (unindented) [inline src/components/MessageBar/message-bar.css]: You could alias these in the `colors.css` file, and have something like `$messageBarColor__error` in this file.

File: src/stories/MessageBar.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    </div>
+  ))
+  .add('with a type', () => (
+    <div>
+      <MessageBar ariaLabel="Error describing what went wrong" type={MessageBarType.WARNING}>
+        Watch out, something unexpected might happen.
+      </MessageBar>
+    </div>
+  ))
+  .add('with an action', () => {
```
Comment (unindented) [inline src/stories/MessageBar.tsx]: Could we have another story with an icon action, like in Zeplin? Just to check that the spacing is all correct.

File: src/stories/gutters.css [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+$gutterWidth__xxLarge: 2.4rem;
+$gutterWidth__xLarge: 2rem;
+$gutterWidth__large: 1.6rem;
+$gutterWidth__medium: 1.2rem;
+$gutterWidth__small: 0.8rem;
+$gutterWidth__xSmall: 0.4rem;
```
Comment (unindented) [inline src/stories/gutters.css]: Why is this file here?

File: src/components/MessageBar/message-bar.css (hunk 3 above)
Comment (richardtagger) [reply src/components/MessageBar/message-bar.css]: mmm I like the locality of putting the message color in this css file. Do we really want styling info for this component in colors.css?

File: src/stories/gutters.css (hunk 5 above)
Comment (richardtagger) [reply src/stories/gutters.css]: no idea

File: src/components/MessageBar/index.tsx (hunk 1 above)
Comment (unindented) [reply src/components/MessageBar/index.tsx]: Feel free to document it. I picked it up by looking at all the existing components, and the tests for the existing components. They all check that `className` can be passed.

File: src/components/MessageBar/message-bar.css (hunk 3 above)
Comment (unindented) [reply src/components/MessageBar/message-bar.css]: It's not the styling info. It's providing semantic names to colors. All component frameworks do this: Fabric, Bootstrap, Foundation, etc. Nobody litters their component CSS files with random color names, but with semantic names.

File: src/stories/gutters.css (hunk 5 above)
Comment (unindented) [reply src/stories/gutters.css]: Maybe it was an accidental drag-drop on your editor? Feel free to kill it.

File: src/components/MessageBar/message-bar.css (hunk 3 above)
Comment (unindented) [reply src/components/MessageBar/message-bar.css]: See for example https://github.com/twbs/bootstrap/blob/9a1d81f154f4391f9fea47fc26cd2fa864d6ae8a/scss/_buttons.scss#L81-L82

They create names for the colors, and then use the names.

File: src/components/MessageBar/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+## Examples
+
+### Basic usage
+
+```js
+const myIconComponent = <Icon icon="lock" />
+<MessageBar
+  ariaLabel="Warning that your post will be visible to some outside your network"
+  type={MessageBarType.Warning}
```
Comment (unindented) [inline src/components/MessageBar/README.md]: This'd be `MessageBarType.WARNING`.

File: src/components/MessageBar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  INFO: 'info',
+  WARNING: 'warning',
+  ERROR: 'error',
+});
+export type MessageBarType = Enum<typeof MessageBarType>;
+
+export interface MessageBarProps extends NestableBaseComponentProps {
+  ariaLabel: string;
+  type?: MessageBarType;
+  actions?: JSX.Element;
```
Comment (unindented) [inline src/components/MessageBar/index.tsx]: This one should be `React.ReactNode` too.

File: src/components/MessageBar/index.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  children: React.ReactNode[] | string;
```
Comment (unindented) [inline src/components/MessageBar/index.tsx]: `React.ReactNode` includes `string`... Have you tried just doing `children: React.ReactNode`, without the array?

Comment (richardtagger) [reply src/components/MessageBar/index.tsx]: i just did what you said before.... i can change it to that

**Comment by richardtagger** [comment]
ok - addressed all the things
